src/mui/input/FileInput.js

```diff
@@ -0,0 +1,128 @@
+import React, { Component, PropTypes } from 'react';
+import Dropzone from 'react-dropzone';
+import DroppedFileField from '../field/DroppedFileField';
+import FileField from '../field/FileField';
+
+const defaultStyle = {
+    dropZone: {
+        background: '#efefef',
+        cursor: 'pointer',
+        padding: '1rem',
+        textAlign: 'center',
+        color: '#999',
+    },
+    previewContainer: {
+        position: 'relative',
+        display: 'inline-block',
+    },
+};
+
+class FileInput extends Component {
+    constructor(props) {
+        super(props);
+
+        let files = props.record[props.source] || [];
+        if (!Array.isArray(files)) {
+            files = [files];
+        }
+
+        this.state = {
+            files: files.map(f => ({
+                ...f,
+                dropped: false,
+            })),
+        };
+    }
+
+    onDrop = (files) => {
+        const updatedFiles = [
+            ...this.state.files,
+            ...files.map(f => ({
+                title: f.name,
+                url: f.preview,
+                dropped: true,
+            })),
+        ];
+
+        this.setState({ files: updatedFiles });
```

Comment:
Not sure you need it, since `input.onChange` will add the new files to `input.value`